src/System.Reflection.Metadata/tests/Metadata/TypeSystem/AssemblyReferenceTests.cs

```diff
@@ -0,0 +1,115 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Linq;
+using Xunit;
+
+namespace System.Reflection.Metadata.Tests
+{
+    public class AssemblyReferenceTests
+    {
+        [Fact]
+        public void ValidateAssemblyNameForSingleAssemblyReference()
+        {
+            var reader = MetadataReaderTests.GetMetadataReader(WinRT.Lib, options: MetadataReaderOptions.ApplyWindowsRuntimeProjections);
+            var handle = reader.AssemblyReferences.Skip(3).First();
+            var assemblyRef = reader.GetAssemblyReference(handle);
+
+            var assemblyName = assemblyRef.GetAssemblyName();
+
+            // Name
+            var refName = reader.GetString(assemblyRef.Name);
+            Assert.Equal("System.Runtime", assemblyName.Name);
+            Assert.Equal(refName, assemblyName.Name);
+
+            // Version
+            var refVersion = assemblyRef.Version;
+            Assert.Equal(new Version(4, 0, 0, 0), assemblyName.Version);
+            Assert.Equal(refVersion, assemblyName.Version);
+
+            // Culture
+            Assert.Equal(null, assemblyName.CultureName);
+
+            // PublicKey
+            var refPublicKey = reader.GetBlobBytes(assemblyRef.PublicKeyOrToken);
+            Assert.Equal(new byte[] { 0xB0, 0x3F, 0x5F, 0x7F, 0x11, 0xD5, 0x0A, 0x3A }, assemblyName.GetPublicKey());
+            Assert.Equal(refPublicKey, assemblyName.GetPublicKey());
+            
+            // HashAlgorithm
+            var refHashAlgorithm = Configuration.Assemblies.AssemblyHashAlgorithm.None;
```

Comment:
No need to create the extra variable. You could just do: `Assert.Equal(Configuration.Assemblies.AssemblyHashAlgorithm.None, assemblyName.HashAlgorithm);`